src/sonic_ax_impl/mibs/ietf/rfc3433.py

```diff
@@ -425,7 +425,9 @@ def update_xcvr_dom_data(self):
             # get transceiver sensors from transceiver dom entry in STATE DB
             transceiver_dom_entry_data = Namespace.dbs_get_all(self.statedb, mibs.STATE_DB, transceiver_dom_entry)
 
-            if not transceiver_dom_entry_data:
+            transceiver_info_entry_data = Namespace.dbs_get_all(self.statedb, mibs.STATE_DB, mibs.transceiver_info_table(interface))
```

Comment:
How about move this line to line 423? In that case, if the type is RJ45, we don't need to read dom data from DB.